plugin/src/py/android_screenshot_tests/device_name_calculator.py

```diff
@@ -0,0 +1,70 @@
+from . import common
+import re
+
+
+class DeviceNameCalculator:
+    def __init__(self):
+        pass
+
+    def name(self):
+        api_version_text = self._api_version_text()
+        play_services_text = self._play_services_text()
+        screen_density_text = self._screen_density_text()
+        screen_size_text = self._screen_size_text()
+
+        return "{0}_{1}_{2}_{3}".format(api_version_text, play_services_text,
+                                        screen_density_text, screen_size_text)
+
+    def _screen_density_text(self):
+        density = self._screen_density()
+
+        try:
+            density = int(density)
+        except:
+            return 'UNKNOWN'
```

Comment:
Instead of defaulting to UNKNOWN I would rather let this exception cause an error. `_screen_density` only looks for a contiguous string of numbers, so this _should_ never fail. In case it does, we shouldn't continue, especially if `None` was returned.